usaspending_api/etl/subaward_etl.py

```diff
@@ -135,17 +147,25 @@ def load_subawards(submission_attributes, db_cursor):
                 d2_empty_count += 1
             continue
 
+        agency = get_valid_awarding_agency(row)
+
+        if not agency:
+            logger.warn("Subaward number {} cannot find matching agency with toptier code {} and subtier code {}".format(row['subaward_num'], row['awarding_agency_code'], row['awarding_sub_tier_agency_c']))
+            continue
+
         # Find the award to attach this sub-award to
         # We perform this lookup by finding the Award containing a transaction with
         # a matching fain and submission. If this fails, try submission and uri
         if row['fain'] and len(row['fain']) > 0:
-            award = Award.objects.filter(transaction__submission=submission_attributes,
+            award = Award.objects.filter(awarding_agency=agency,
+                                         transaction__submission=submission_attributes,
                                          transaction__assistance_data__isnull=False,
                                          transaction__assistance_data__fain=row['fain']).distinct().first()
 
         # Couldn't find a match on FAIN, try URI if it exists
         if not award and row['uri'] and len(row['uri']) > 0:
-            award = Award.objects.filter(transaction__submission=submission_attributes,
+            award = Award.objects.filter(awarding_agency=agency,
```

Comment:
Same question as above...would it be useful/more readable to query the `uri` and `fain` on the award itself instead of digging into `transaction__assistance_data`?